### PR TITLE
Add Spacey keyboard firmware

### PR DESCRIPTION
**Pull request by vanillakeyboards**
## Description

Add new board from p3d and vanilla keyboards

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/spacey/readme.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# spacey
+
+![spacey](imgur.com image replace me!)
+
+*A short description of the keyboard/project*
+
+* Keyboard Maintainer: [vanilla](https://github.com/yourusername)
+* Hardware Supported: *The PCBs, controllers supported*
+* Hardware Availability: *Links to where you can find this hardware*
+
```
Comment (drashna) [inline keyboards/spacey/readme.md:10]: please fill in this information.

File: keyboards/spacey/spacey.h [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+/* This is a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT( \
```
Comment (drashna) [inline keyboards/spacey/spacey.h:29]: This looks like it's an `ortho_5x14` Board?

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/spacey/spacey.h (hunk 2 above)
Comment (vanillakeyboards) [reply keyboards/spacey/spacey.h:29]: yes but no, new imgur link is over in the readme

File: keyboards/spacey/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#define XXX KC_NO
```
Comment (drashna) [inline keyboards/spacey/keymaps/default/keymap.c:17]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/spacey/config.h [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+
+/* define if matrix has ghost (lacks anti-ghosting diodes) */
+//#define MATRIX_HAS_GHOST
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
```
Comment (drashna) [inline keyboards/spacey/config.h:106]:
```suggestion
```

File: keyboards/spacey/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+void encoder_update_user(uint8_t index, bool clockwise){
+    if (index == 0){
+        if(clockwise){
+            tap_code(KC_VOLD);
+        } else{
+            tap_code(KC_VOLU);
+        }
+    }
+    return;
+}
```
Comment (drashna) [inline keyboards/spacey/keymaps/default/keymap.c:62]:
```suggestion
```

File: keyboards/spacey/spacey.c [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "spacey.h"
```
Comment (drashna) [inline keyboards/spacey/spacey.c:17]: This lets you set a default behavior that can be overridden, and will work for the configurator. 

```suggestion
#include "spacey.h"

void encoder_update_kb(uint8_t index, bool clockwise){
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if(clockwise) {
            tap_code(KC_VOLD);
        } else {
            tap_code(KC_VOLU);
        }
    }
    return;
} 

```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/spacey/readme.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
+# spacey
+
+![spacey](https://imgur.com/a/ljJw87d)
```
Comment (fauxpark) [inline keyboards/spacey/readme.md:3]:
```suggestion
![spacey](https://i.imgur.com/Dzon2hbh.jpeg)
```
Needs to be a direct image link.

File: keyboards/spacey/readme.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+*A short description of the keyboard/project*
```
Comment (fauxpark) [inline keyboards/spacey/readme.md:5]: Please fill this out.

File: keyboards/spacey/readme.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+* Keyboard Maintainer: [vanilla](https://github.com/vanillakeyboards)
+* Hardware Supported: *Elite C, Spacey PCB plates and acrylic drop_in case*
+* Hardware Availability: *https://p3dstore.com/collections/ended-group-buys/products/spacey?variant=37742593147064 and https://keeb.io/products/elite-c-low-profile-version-usb-c-pro-micro-replacement-atmega32u4*
```
Comment (fauxpark) [inline keyboards/spacey/readme.md:9]:
```suggestion
* Hardware Supported: Elite C, Spacey PCB plates and acrylic drop_in case
* Hardware Availability: https://p3dstore.com/collections/ended-group-buys/products/spacey?variant=37742593147064 and https://keeb.io/products/elite-c-low-profile-version-usb-c-pro-micro-replacement-atmega32u4
```

File: keyboards/spacey/spacey.c [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+void encoder_update_kb(uint8_t index, bool clockwise){
```
Comment (fauxpark) [inline keyboards/spacey/spacey.c:19]:
```suggestion
bool encoder_update_kb(uint8_t index, bool clockwise){
```

File: keyboards/spacey/spacey.c [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    if (!encoder_update_user(index, clockwise)) { return false; }
+    if (index == 0) {
+        if(clockwise) {
+            tap_code(KC_VOLD);
+        } else {
+            tap_code(KC_VOLU);
+        }
+    }
+    return;
```
Comment (fauxpark) [inline keyboards/spacey/spacey.c:28]:
```suggestion
    return true;
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/spacey/readme.md (hunk 8 above)
Comment (vanillakeyboards) [reply keyboards/spacey/readme.md]: Just dropped a fresh commit!

**Review by fauxpark — APPROVED** [review]

**Comment by vanillakeyboards** [comment]
Is there anything left on my end that I can do to help get this approved to merge in?

**Comment by drashna** [comment]
Thanks!

**Pull request merged**